decidim-core/db/migrate/20250221085317_add_user_type_to_action_logs.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class AddUserTypeToActionLogs < ActiveRecord::Migration[7.0]
+  class User < ApplicationRecord
+    self.table_name = :decidim_users
+    self.inheritance_column = nil # disable the default inheritance
+
+    default_scope { where(type: "Decidim::User") }
+  end
+
+  class ActionLog < ApplicationRecord
+    self.table_name = :decidim_action_logs
+  end
+
+  def change
+    add_column :decidim_action_logs, :user_type, :string
+
+    ActionLog.find_each do |log|
+      user = User.find_by(id: log.decidim_user_id) if log.decidim_user_id.present?
+      log.update!(
+        user_type: user.is_a?(User) ? user.type : "Decidim::Api::ApiUser",
+        decidim_user_id: log.decidim_user_id
+      )
+    end
```

Comment:
This does not work for the `down` migration. All migrations should be reversable.

Also, this is very slow with a large user base and we should avoid such slow migrations.

This type of DB changes should be passed directly to the database server, as that is very fast.